src/roles/pulp/tasks/main.yaml

```
@@ -128,41 +130,34 @@
     name: pulpcore-manager-migrate
     image: "{{ pulp_api_image }}"
     command: pulpcore-manager migrate --noinput
+    detach: false
     network: host
     secrets:
       - 'pulp-symmetric-key,type=mount,target=/etc/pulp/certs/database_fields.symmetric.key'
       - 'pulp-db-password,type=env,target=PULP_DATABASES__default__PASSWORD'
     env: "{{ pulp_settings_database_env }}"
 
-- name: Start the Pulp API services
+- name: Start Pulp services
   ansible.builtin.systemd:
-    name: pulp-api
-    enabled: true
-    state: started
-
-- name: Wait for Pulp API service to be accessible
-  ansible.builtin.wait_for:
-    host: "{{ ansible_fqdn}}"
-    port: 24817
-    timeout: 300
-
-- name: Start the Pulp Content services
-  ansible.builtin.systemd:
-    name: pulp-content
-    enabled: true
-    state: started
-
-- name: Wait for Pulp Content service to be accessible
-  ansible.builtin.wait_for:
-    host: "{{ ansible_fqdn }}"
-    port: 24816
-    timeout: 600
-
-- name: Start the Pulp Worker service
-  ansible.builtin.systemd:
-    name: pulp-worker
+    name: "{{ item }}"
     enabled: true
     state: started
+  async: 3
```

```suggestion
  async: 30
```

The task runs in a timeout if it's not completed after `async` seconds. Which is what happened here, or at least that's what I assume happened.

---

So `async` is the number of seconds to wait, not the concurrency as I assumed? Interesting.

---

I agree, the fundamentals in Ansible for configuration management just suck. I think Puppet with notifications solved restarts much better than handlers. Granted, it doesn't have anything for async. Many years ago I had hopes for mgmt to solve both well.